index.js

```diff
@@ -15,6 +15,7 @@ function getStream(inputStream, opts) {
 
 	var p = new Promise(function (resolve, reject) {
 		stream = bufferStream(opts);
+		inputStream.on('error', reject);
```

Comment:
let's attach the buffered value up to the point of rejection to the Error object:

```js
inputStream.on('error', err => {
  if (err) { // null check
    err.buffer = stream.getBufferedValue();
  }
  reject(err);
});
```

Make sure to add that to the docs and update your test to verify the behavior.